### PR TITLE
Add Splice v1.5.12

### PR DESCRIPTION
**Pull request by danielbayley**
Splice cloud platform for music.

Signed-off-by: Daniel Bayley daniel.bayley@me.com

### CODE REVIEW

File: Casks/splice.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  url 'https://spliceosx.s3.amazonaws.com/Splice.dmg'
+  name 'Splice'
+  homepage 'http://splice.com'
+  license :unknown
+
+  installer :script => 'Splice Installer.app/Contents/MacOS/Splice Installer',
+            :args => [ '-q' ],
+            :sudo => false
+  postflight do
```
Comment (vitorgalvao) [inline Casks/splice.rb]: Don’t kill it. Let it run. This is something we’ll have to fix later, in a better way.

File: Casks/splice.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+cask :v1 => 'splice' do
+  version :latest
+  sha256 :no_check
+
+  url 'https://spliceosx.s3.amazonaws.com/Splice.dmg'
```
Comment (vitorgalvao) [inline Casks/splice.rb]: Needs a `# amazonaws.com is the official download host per the vendor homepage` comment above this line ([example](https://github.com/caskroom/homebrew-cask/blob/cb487ad405f74cb1f26b93e3f3e8412c82a22e34/Casks/0ad.rb#L5)).

File: Casks/splice.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  name 'Splice'
+  homepage 'http://splice.com'
+  license :unknown
```
Comment (vitorgalvao) [inline Casks/splice.rb]: `:gratis`.

**Comment by vitorgalvao** [comment]
Needs squashing.

File: Casks/splice.rb (hunk 1 above)
Comment (danielbayley) [inline Casks/splice.rb]: So homebrew-cask is causing it to launch?
There's no 'proper' way to stop that now then?

File: Casks/splice.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  license :gratis
+
+  installer :script => 'Splice Installer.app/Contents/MacOS/Splice Installer',
+            :args => [ '-q' ],
+            :sudo => false
+
+  uninstall :quit => 'com.splice.Splice',
+            :delete => '/Applications/Splice.app'
+
+  zap :delete => '~/Library/*/com.splice*'
```
Comment (vitorgalvao) [inline Casks/splice.rb]: I doubt this’ll produce good results. Please input each path.

Comment (danielbayley) [inline Casks/splice.rb]: Would changing it to `~/Library/*/com.splice.Splice*` do? It picks up the following just fine.
```
~/Library/Application Support/com.splice.Splice
~/Library/Caches/com.splice.Splice
~/Library/Caches/com.splice.Splice-Installer
~/Library/Preferences/com.splice.Splice.plist
```

Comment (vitorgalvao) [inline Casks/splice.rb]: Even if it works, it’s too obscure. Users should have a sense of what is being removed.

File: Casks/splice.rb (hunk 1 above)
Comment (vitorgalvao) [inline Casks/splice.rb]: No, it isn’t homebrew-cask launching it, but the installer. What we don’t have is a proper way to handle killing it.

Comment (danielbayley) [inline Casks/splice.rb]: I suppose there needs to be something like `suppress_application_launch` for these cases, similar to `suppress_move_to_applications`.

**Comment by vitorgalvao** [comment]
Need me to do the squashing?

**Comment by danielbayley** [comment]
That would be the easy option, but i'd like to know the best way of doing that anyway, preferably in Tower.

**Comment by vitorgalvao** [comment]
As for Tower, I have no idea as I don’t use any GUI client for git (nor have I ever used one for a significant amount of time).

As for the CLI, [this](http://davidwalsh.name/squash-commits-git) should explain it. This one in particular is a bit of a mess, though, as you have commits from a previous PR, which is messing up even my pulling of the patch.

**Comment by danielbayley** [comment]
> This one in particular is a bit of a mess, though, as you have commits from a previous PR, which is messing up even my pulling of the patch.

Yeah not sure why; I thought it was odd that it lists '7 commits', most of them unrelated. I'm thinking just scrap this branch and do it again?

The `splice.rb` is safely stashed.

**Comment by vitorgalvao** [comment]
> I'm thinking just scrap this branch and do it again?

Sure.

**Pull request closed without merging**